lib/domain/models/card_execution.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:equatable/equatable.dart';
+import 'package:memo/data/database_repository.dart';
+import 'package:memo/domain/enums/card_difficulty.dart';
+import 'package:memo/domain/models/card_block.dart';
+import 'package:meta/meta.dart';
+
+/// Representation of the exact history (immutable) of a `Card` execution
+@immutable
+class CardExecution extends Equatable {
+  CardExecution({
+    required this.started,
+    required this.finished,
+    required this.question,
+    required this.answer,
+    required this.answeredDifficulty,
+  })   : assert(started.isBefore(finished)),
+        assert(question.isNotEmpty),
+        assert(answer.isNotEmpty);
+
+  final DateTime started;
```

Comment:
IMO `startedAt` and `finishedAt` are better. Just `started` and `finished` may confuse with a boolean like `isFinished`

Reply:
I think `startedAt` comes from a Swift background where function signatures can have the full argument like `myFunc(startedAt startedDate:)`. Just like we don't name an `int index` as `int indexAt`. This is a personal opinion and while I agree, I think it should be `startDate` instead of suffixing with `at`